hilti/toolchain/src/compiler/cxx/linker.cc

```diff
@@ -63,8 +68,11 @@ void cxx::Linker::finalize() {
             if ( ! c.declare_only )
                 unit.add(c.callee);
 
-            for ( const auto& t : c.aux_types )
+            for ( auto t : c.aux_types ) {
+                // For the linker file forward declarations are sufficient.
+                t.forward_decl = true;
```

Comment:
The issue somewhere in this patch is still that types without hooks lead to a different linker metadata than files with hooks. While for the former we emit a forward declaration, for the latter (e.g., `Pointer` here) we emit a full declaration of the type, but do not include its own type dependencies.